From 30b2528054e6627a7124ac04cb018356ef23d864 Mon Sep 17 00:00:00 2001
From: Petr Portnov <mrjarviscraft@gmail.com>
Date: Mon, 2 Sep 2024 22:25:33 +0300
Subject: [PATCH 1/1] build: reduce hardcode in `asio_path`

Currently, `asio_path` variable value is concatenated with `/asio/include`
to specify the path to custom `asio` installation.
The problem is that this is too strict as some distros (namely NixOS)
may have the `include` directory with a differently named parent.
Thus this change minimizes the hardcoded part of the path to make it more flexible.

Signed-off-by: Petr Portnov <mrjarviscraft@gmail.com>
---
 meson.build       | 2 +-
 meson_options.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index c9e0a2d..c01eb8e 100644
--- a/meson.build
+++ b/meson.build
@@ -74,7 +74,7 @@ endif
 #
 #  Setup additional include header dirs
 #
-asio_inc = get_option('asio_path') / 'asio' / 'include'
+asio_inc = get_option('asio_path') / 'include'
 message ('ASIO library: ' + asio_inc)
 
 openvpn3_core_inc = get_option('openvpn3_core_path')
diff --git a/meson_options.txt b/meson_options.txt
index d9cf02e..43e301e 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -26,7 +26,7 @@ option('debug_options', type: 'boolean', value: false,
 #
 #  Build environment and related build time options
 #
-option('asio_path', type: 'string', value: './vendor/asio',
+option('asio_path', type: 'string', value: './vendor/asio/asio',
        description: 'Path to the ASIO header files')
 
 option('openvpn3_core_path', type: 'string', value: './openvpn3-core',
-- 
2.43.0

